src/pybind/mgr/cephadm/serve.py

```diff
@@ -713,6 +714,40 @@ def matches_network(host):
             )
             return False
 
+        def has_interface_for_vip(host: str, sspec: ServiceSpec) -> bool:
+            # make sure the host has an interface that can
+            # actually accomodate the VIP
+            if not sspec or sspec.service_type != 'ingress':
+                return True
+            ingress_spec = cast(IngressSpec, sspec)
+            virtual_ips = []
+            if ingress_spec.virtual_ip:
+                virtual_ips.append(ingress_spec.virtual_ip)
+            elif ingress_spec.virtual_ips_list:
+                virtual_ips = ingress_spec.virtual_ips_list
+            for vip in virtual_ips:
+                found = False
+                bare_ip = str(vip).split('/')[0]
+                for subnet, ifaces in self.mgr.cache.networks.get(host, {}).items():
+                    if ifaces and ipaddress.ip_address(bare_ip) in ipaddress.ip_network(subnet):
```

Comment:
I feel like I keep seeing this same loop. Not for this PR, per se, but maybe we should try to encapulate this common logic in a function, possibly one that takes a predicate method for filtering.

Reply:
made a tracker for this so we can cover it in a follow up https://tracker.ceph.com/issues/62327